### PR TITLE
Using go modules for docker

### PR DESCRIPTION
**Pull request by contrun**
Should have fixed https://github.com/filecoin-project/go-filecoin/issues/2494

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,8 @@ ENV SRC_DIR /go/src/github.com/filecoin-project/go-filecoin
 COPY . $SRC_DIR
 
 # Build the thing.
-RUN cd $SRC_DIR \
+RUN export GO111MODULE=on \
```
Comment (rkowalick) [inline Dockerfile]: Can you use the `ENV` command as above on line 11?

File: Dockerfile.faucet [hunk 2]
```diff
@@ -15,7 +15,8 @@ ENV FILECOIN_USE_PRECOMPILED_BLS_SIGNATURES true
 COPY . $SRC_DIR
 
 # Build faucet
-RUN cd $SRC_DIR \
+RUN export GO111MODULE=on \
```
Comment (rkowalick) [inline Dockerfile.faucet]: Same comment as above

File: Dockerfile.genesis [hunk 3]
```diff
@@ -13,7 +13,8 @@ ENV SRC_DIR /go/src/github.com/filecoin-project/go-filecoin
 COPY . $SRC_DIR
 
 # Build genesis server
-RUN cd $SRC_DIR \
+RUN export GO111MODULE=on \
```
Comment (rkowalick) [inline Dockerfile.genesis]: Same comment as above

**Comment by rkowalick** [comment]
@contrun Thanks for the contribution! I left a few comments which should be simple to address. Happy to merge after approval from @ognots .

**Comment by ognots** [comment]
+1 on feedback from @rkowalic, will approve when implemented.
thanks for the fix!

**Comment by contrun** [comment]
@ognots  Implemented.

**Comment by rkowalick** [comment]
@ognots The build is failing and I'm not sure why

**Comment by contrun** [comment]
@rkowalick may be this comment is related https://github.com/filecoin-project/go-filecoin/pull/2473#issuecomment-480302294

**Pull request merged**